### PR TITLE
Use testcontainers instead of H2

### PR DESCRIPTION
**Pull request by petertrr**
### What's done:
* Added dependencies
* Converted DatabaseTest.kt
* Configured properties

### TODO:
* [x] Share DB container among tests
* [x] Extract dependencies to buildSrc

### CODE REVIEW

File: save-backend/src/test/kotlin/org/cqfn/save/backend/utils/DatabaseTestBase.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+/**
+ * A base class giving access to a database instance. Is used instead of `@Testcontainers` annotations:
+ * we manually start container here, and DB connection can be used in all tests inheriting this class.
+ */
+open class DatabaseTestBase {
+    companion object {
+        @Container
+        @JvmStatic
+        val dbContainer: JdbcDatabaseContainer<*> = MySQLContainerProvider()
+            .newInstance("8.0")
```
Comment (akuleshov7) [inline save-backend/src/test/kotlin/org/cqfn/save/backend/utils/DatabaseTestBase.kt:19]: 8.0.20

File: save-backend/src/test/kotlin/org/cqfn/save/backend/utils/DatabaseTestBase.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            .withExposedPorts(3306)
```
Comment (akuleshov7) [inline save-backend/src/test/kotlin/org/cqfn/save/backend/utils/DatabaseTestBase.kt:20]: need to add test port to docs

File: save-backend/src/test/kotlin/org/cqfn/save/backend/utils/DatabaseTestBase.kt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            .withDatabaseName("db_example")
```
Comment (akuleshov7) [inline save-backend/src/test/kotlin/org/cqfn/save/backend/utils/DatabaseTestBase.kt:21]: save_db_test

**Review by akuleshov7 — APPROVED** [review]
lgtm

File: save-deploy/README.md [hunk 4]
```diff
@@ -23,6 +23,7 @@ Deployment is performed on server via docker swarm or locally via docker-compose
 ## Ports allocation
 | port | descritpion |
 | ---- | ----------- |
+} 3306 | database (locally) |
```
Comment (aktsay6) [inline save-deploy/README.md:26]: Guess that we need to delete h2 console here, since we are not using it

Comment (petertrr) [reply save-deploy/README.md:26]: right, thx

**Review by aktsay6 — APPROVED** [review]
lgtm

**Pull request merged**